src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationClaimantOfHearingHandler.java

```diff
@@ -72,31 +65,67 @@ public List<CaseEvent> handledEvents() {
 
     @Override
     public Map<String, String> addProperties(final CaseData caseData) {
-        String hourMinute = caseData.getHearingTimeHourMinute();
         String reference;
-        int hours = Integer.parseInt(hourMinute.substring(0, 2));
-        int minutes = Integer.parseInt(hourMinute.substring(2, 4));
-        LocalTime time = LocalTime.of(hours, minutes, 0);
-        if (caseData.getSolicitorReferences() == null
-            || caseData.getSolicitorReferences().getApplicantSolicitor1Reference() == null) {
-            reference = "";
+        String legacyCaseRef = caseData.getLegacyCaseReference();
+        String hearingDate = NotificationUtils.getFormattedHearingDate(caseData);
+        String hearingTime = NotificationUtils.getFormattedHearingTime(caseData);
+        if (!isApplicantLip(caseData)) {
+            if (caseData.getSolicitorReferences() == null
```

Comment:
we can simplify this method
something like this:

   public Map<String, String> addProperties(final CaseData caseData) {
        String reference;
        String legacyCaseRef = caseData.getLegacyCaseReference();
        String hearingDate = NotificationUtils.getFormattedHearingDate(caseData);
        String hearingTime = NotificationUtils.getFormattedHearingTime(caseData);
        var map =  new HashMap<>(Map.of(
            CLAIM_REFERENCE_NUMBER,
            legacyCaseRef,
            HEARING_DATE,
            hearingDate,
            HEARING_TIME,
            hearingTime));

        if (!isApplicantLip(caseData)) {
            String reference = "";
            if (!(caseData.getSolicitorReferences() == null
                || caseData.getSolicitorReferences().getApplicantSolicitor1Reference() == null)) {
                reference = caseData.getSolicitorReferences().getApplicantSolicitor1Reference();
            } 
            map.put(caseData.getHearingFee() == null ? "£0.00" : String.valueOf(caseData.getHearingFee().formData()));
            map.put(caseData.getHearingDueDate() == null ? "" :
                        caseData.getHearingDueDate().format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
            map.put(HEARING_DUE_DATE);
            map.put(caseData.getHearingDueDate() == null ? "" :
                        caseData.getHearingDueDate().format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
            map.put(CLAIMANT_REFERENCE_NUMBER, reference);
            
        } 
        return map;
        
    }
    
    What do you think?

Reply:
Very true! it is not very readable and can be simplified, I'll change it.